front/src/modules/ui/board/components/BoardHeader.tsx

```diff
@@ -16,31 +24,73 @@ export type BoardHeaderProps<SortField> = ComponentProps<'div'> & {
     ViewBarProps<SortField>,
     | 'availableSorts'
     | 'defaultViewName'
-    | 'onViewsChange'
+    | 'onViewChange'
     | 'onViewSubmit'
     | 'scopeContext'
   >;
 
 export function BoardHeader<SortField>({
   onStageAdd,
-  onViewsChange,
+  onViewChange,
   onViewSubmit,
   scopeContext,
-  availableSorts,
-  defaultViewName,
+  ...props
 }: BoardHeaderProps<SortField>) {
+  const tableScopeId = useContextScopeId(scopeContext);
+
+  const currentViewId = useRecoilScopedValue(
+    currentViewIdScopedState,
+    scopeContext,
+  );
+  const canPersistBoardCardFields = useRecoilValue(
+    canPersistBoardCardFieldsScopedFamilySelector([
+      tableScopeId,
+      currentViewId,
+    ]),
+  );
+  const [boardCardFields, setBoardCardFields] = useRecoilScopedState(
+    boardCardFieldsScopedState,
+    scopeContext,
+  );
+  const [savedBoardCardFields, setSavedBoardCardFields] = useRecoilState(
+    savedBoardCardFieldsFamilyState(currentViewId),
+  );
+
+  const handleViewBarReset = () => setBoardCardFields(savedBoardCardFields);
+
+  const handleViewSelect = useRecoilCallback(
+    ({ set, snapshot }) =>
+      async (viewId: string) => {
+        const savedBoardCardFields = await snapshot.getPromise(
+          savedBoardCardFieldsFamilyState(viewId),
+        );
+        set(boardCardFieldsScopedState(tableScopeId), savedBoardCardFields);
+      },
+    [tableScopeId],
+  );
+
+  async function handleViewSubmit() {
+    if (canPersistBoardCardFields) {
+      setSavedBoardCardFields(boardCardFields);
+    }
+
+    await onViewSubmit?.();
+  }
+
   return (
     <RecoilScope SpecificContext={DropdownRecoilScopeContext}>
       <ViewBar
-        availableSorts={availableSorts}
-        defaultViewName={defaultViewName}
-        onViewsChange={onViewsChange}
-        onViewSubmit={onViewSubmit}
+        {...props}
```

Comment:
I am still not a big fan of this { ...props } pattern, I always have a hard time understanding what is passed within it :)

Reply:
Ok, I removed it.